script.js

```diff
@@ -1,4 +1,223 @@
+      if (cardName == 1) {
+        cardName = "ace";
+      } else if (cardName == 11) {
+        cardName = "jack";
+      } else if (cardName == 12) {
+        cardName = "queen";
+      } else if (cardName == 13) {
+        cardName = "king";
+      }
+
+      // Create a new card with the current name, suit, and rank
+      var card = {
+        name: cardName,
+        suit: currentSuit,
+        rank: rankCounter,
+      };
+
+      // Add the new card to the deck
+      cardDeck.push(card);
+
+      // Increment rankCounter to iterate over the next rank
+      rankCounter += 1;
+    }
+
+    // Increment the suit index to iterate over the next suit
+    suitIndex += 1;
+  }
+
+  // Return the completed card deck
+  return cardDeck;
+};
+
+// Copied random index generator
+var getRandomIndex = function (max) {
+  return Math.floor(Math.random() * max);
+};
+
+// Copied shuffle deck function
+var shuffleCards = function (cardDeck) {
+  // Loop over the card deck array once
+  var currentIndex = 0;
+  while (currentIndex < cardDeck.length) {
+    // Select a random index in the deck
+    var randomIndex = getRandomIndex(cardDeck.length);
+    // Select the card that corresponds to randomIndex
+    var randomCard = cardDeck[randomIndex];
+    // Select the card that corresponds to currentIndex
+    var currentCard = cardDeck[currentIndex];
+    // Swap positions of randomCard and currentCard in the deck
+    cardDeck[currentIndex] = randomCard;
+    cardDeck[randomIndex] = currentCard;
+    // Increment currentIndex to shuffle the next pair of cards
+    currentIndex += 1;
+  }
+  // Return the shuffled deck
+  return cardDeck;
+};
+
+// Copied initialisation of the shuffled card deck before the game starts.
+var deck = shuffleCards(makeDeck());
+//Inital mode
+var gameMode = `Gamestart`;
+
+// Blackjack gameplay
+// Need global variable for player and computer hand in the form of an array and a function to deal cards
+var playerHand = [];
+var computerHand = [];
+// var cardDeal = function (hand) {
+//   hand.push(deck.pop());
+// };
+// Need to calculate player and computer's points
+var playerPoints = 0;
+var computerPoints = 0;
+
+// Ace
+var numOfAce = 0;
+
+var pointCalculation = function (hand) {
+  var totalPoints = 0;
+  // To calculate each card's points starting from index 0, 1 ...
+  for (let i = 0; i < hand.length; i += 1) {
+    var cardValue = hand[i];
+    // 2 to 10
+    if (cardValue.rank >= 2 && cardValue.rank <= 10) {
+      totalPoints += cardValue.name;
+    }
+    // jack queen king
+    if (cardValue.rank >= 11 && cardValue.rank <= 13) {
+      totalPoints += 10;
+    }
+    // Ace(11)
+    else if (cardValue.rank == 1) {
+      totalPoints += 11;
+      numOfAce += 1;
+      if (playerHand.length > 2 || numOfAce == 2) {
+        totalPoints -= 1;
+      }
+      if (playerHand.length > 2 || numOfAce > 2) {
+        totalPoints -= 10;
+      }
```

Comment:
For lines 117 to 122, if the player has >2 cards in his hand and wants his ace to have a value of 11, he is unable to do so.